pkg/kubelet/kubelet.go

```diff
@@ -2042,11 +2042,18 @@ func (kl *Kubelet) HandlePodRemoves(pods []*v1.Pod) {
 // HandlePodReconcile is the callback in the SyncHandler interface for pods
 // that should be reconciled.
 func (kl *Kubelet) HandlePodReconcile(pods []*v1.Pod) {
+	start := kl.clock.Now()
 	for _, pod := range pods {
 		// Update the pod in pod manager, status manager will do periodically reconcile according
 		// to the pod manager.
 		kl.podManager.UpdatePod(pod)
 
+		// Reconcile Pod "Ready" condition if necessary. Trigger sync pod for reconciliation.
+		if status.NeedToReconcilePodReadiness(pod) {
```

Comment:
Question for the authors, why did we *only* trigger the pod worker (which is what drives status manager) here?

A reconcile is triggered when status is mutated from the API server - was this done to try and minimize the number of status sync events for a running pod?

Reply:
I'm not sure @freehan is paying attention to k/k any more :(